### DISCUSSION TITLE
How to Disable Creating A Script File For Each New Editor

### DISCUSSION THREAD

**Discussion by teschste-reyrey** [dbeaver/dbeaver#14358] [Q&A]
Whenever I open a new SQL editor, DBeaver creates a new script file in the background, presumably to prevent data loss.  However, I would expect that to be an option, not a requirement.  The new scripts are saved in a single folder, which is never where I want the scripts to end up.

Normally when I create a script, I choose Save (other UI's) or Save As, and specify a filename to save each script where I need it and I am done, all within the interface.  However DBeaver is forcing me to either:

- Rename the script and then go to the file system to move it where I need it.
- Ignore the default script it created, do a Save As, and then periodically go back and delete all of the default scripts it created.

There are a lot of options in DBeaver, so I am hoping there is an option I just missed that will stop it from creating these files automatically.

Thanks!

**Comment by p91paul** [comment]
From the create script button dropdown, there is also an option to create a sql console. A sql console is akin to a script, except it is not linked to a file and cannot be saved. It's perfect for throwaway queries!

**Comment by teschste-reyrey** [comment]
However I am not creating throw away queries.  I am generally creating DDL scripts.  My process is, and has been in every database UI I have ever used, to open a SQL editor, create the script, and then save the script.

When I do that in DBeaver, if I choose the default Save option, I have no choice of where to save the script (the target location will be different for every project), so I have to go to the OS and move it from the single location DBeaver uses to the location I need it, and then rename it.  Conversely, if I choose the Save As option, the script DBeaver automatically created is left behind, so I have to delete it later.

I would expect that there would either be an option to **not** automatically create a script, or when I choose Save As, the script DBeaver automatically created would be moved to the location, and with the name, that I chose in the Save As dialog.  However I cannot find either option, which is the reason for this discussion.

Do I just need to add this as an issue?  Thanks!

**Reply by p91paul** [reply]
That "save as" behavior might be considered a bug, so yes, I suggest you open an issue.
